python/ray/autoscaler/v2/instance_manager/instance_manager.py

```diff
@@ -0,0 +1,296 @@
+from collections import defaultdict
+import copy
+import logging
+import time
+import uuid
+from abc import ABCMeta, abstractmethod
+from typing import Any, Dict, List, Optional
+
+from ray.autoscaler.v2.instance_manager.instance_storage import (
+    InstanceStorage,
+    InstanceUpdatedSubscriber,
+)
+from ray.core.generated.instance_manager_pb2 import (
+    Status,
+    GetInstanceManagerStateReply,
+    Instance,
+    InstanceManagerState,
+    UpdateInstanceManagerStateReply,
+    UpdateInstanceManagerStateRequest,
+    LaunchRequest,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class InstanceManager(metaclass=ABCMeta):
+    """
+    See `InstanceManagerService` in instance_manager.proto
+    """
+
+    @abstractmethod
+    def update_instance_manager_state(
+        self, request: UpdateInstanceManagerStateRequest
+    ) -> UpdateInstanceManagerStateReply:
+        pass
+
+    @abstractmethod
+    def get_instance_manager_state(self) -> GetInstanceManagerStateReply:
+        pass
+
+
+class SimpleInstanceManager(InstanceManager):
```

Comment:
```suggestion
class InMemoryInstanceManager(InstanceManager):
```
?

Reply:
hmm - it's actually the nstance storage part is in-memory. The instance memory is able to work with a persistent instance storage here.